README.md

```diff
@@ -68,10 +68,56 @@ Floe::Workflow::Runner.docker_runner = Floe::Workflow::Runner::Podman.new
 # Or
 Floe::Workflow::Runner.docker_runner = Floe::Workflow::Runner::Kubernetes.new("namespace" => "default", "server" => "https://k8s.example.com:6443", "token" => "my-token")
 
-workflow = Floe::Workflow.load(File.read("workflow.asl"))
+workflow = Floe::Workflow.load("workflow.asl")
 workflow.run!
 ```
 
+### Non-Blocking Workflow Execution
+
+It is also possible to step through a workflow without blocking, and any state which
+would block will return `Errno::EAGAIN`.
+
+```ruby
+require 'floe'
+
+workflow = Floe::Workflow.load("workflow.asl")
+workflow.run_async!
+
+until workflow.end?
+  loop while workflow.step_nonblock == 0 # Step through the workflow while it would not block
+  break
+end
```

Comment:
Maybe

```ruby
class Workflow
  # run_nonblock
  def run_until_block
    loop while step_nonblock == 0 unless end?
  end
end
```

I'm thinking the `until` and the `break` together are just an `if`?

Regardless of what the code looks like, introducing a method to make documentation simpler seems like a good plan.

Reply:
Okay how about instead of `Workflow#run_async!` we have `Workflow#run_nonblock` which behaves similarly to `step_nonblock` but incorporates your loop.